### PR TITLE
Add SGLabel pool

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#5784

Changes proposed in this pull request:
 * Add SGLabel pool

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [x] Make sure any runtime log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

File: cocos2d/core/platform/js.js [hunk 1]
```diff
@@ -975,6 +975,17 @@ Pool.prototype.put = function (obj) {
...
 };
 
+/**
+ * !#en Resize the pool.
+ * !#zh 设置对象池容量。
+ * @method resize
+ */
+Pool.prototype.resize = function (length) {
+    if (length >= 0) {
+        this._pool.length = length;
```
Comment (jareguo) [inline cocos2d/core/platform/js.js:998]: 如果 length > count，这里应该刷新一下 count

**Review by jareguo — CHANGES_REQUESTED** [review]

File: jsb/jsb-label.js [hunk 2]
```diff
@@ -225,6 +225,8 @@ jsbLabel.prototype.getOutlineColor = function() {
     return this._outlineColor || cc.color(255,255,255,255);
 };
 
+jsbLabel.pool = new JS.Pool(0);
```
Comment (jareguo) [inline jsb/jsb-label.js:228]: jsb 上不缓存吗？

Comment (pandamicro) [reply jsb/jsb-label.js:228]: JSB 的适配还没做，之后再加

File: cocos2d/core/platform/js.js (hunk 1 above)
Comment (pandamicro) [reply cocos2d/core/platform/js.js:998]: OK

**Comment by pandamicro** [comment]
尝试了一下实现 JSB 的 label 缓冲池，但是由于 label 状态过于复杂，缓冲池在回收对象过程中的重置操作有太多 JSB 函数调用，性能并没有优势，所以可以略过了

**Review by jareguo — APPROVED** [review]

**Comment by jareguo** [comment]
最后场景加载速度提升了多少？

**Comment by pandamicro** [comment]
没测试场景加载，这个取决于场景中有多少 label 以及是不是重复进入场景（池子中有没有 label），主要是减少 memory footprint，降低垃圾回收的频率，在游戏项目那边验证了

**Comment by zilongshanren** [comment]
@nantas  这个 PR 在编辑器上面有 bug，建议 revert。。。。 @jareguo

**Comment by zilongshanren** [comment]
@pandamicro 我刚刚发现运行时也有 bug。。。

**Comment by jareguo** [comment]
这个 @pandamicro 看看好不好修吧，好修就直接修了，不好修再 revert

**Pull request merged**